src/react/utils.js

```diff
@@ -235,6 +239,33 @@ export function forEachArrayValue(realm: Realm, array: ObjectValue, mapFunc: Fun
   }
 }
 
+export function mapArrayValue(
+  realm: Realm,
+  array: ArrayValue,
+  mapFunc: (element: Value, descriptor: Descriptor) => Value
+): ArrayValue {
+  let lengthValue = Get(realm, array, "length");
+  invariant(lengthValue instanceof NumberValue, "Invalid length on ArrayValue during reconcilation");
```

Comment:
It really cannot be an `AbstractValue`?

Reply:
Nope, it shouldn't be.

Reply:
I've made the invariant a TODO incase this ever comes up. It shouldn't though but at least the invariant makes more sense now.